packages/cubejs-client-core/index.d.ts

```diff
@@ -816,7 +816,7 @@ declare module '@cubejs-client/core' {
     never;
 
   type QueryArrayRecordType<T extends DeeplyReadonly<Query[]>> =
-    T extends readonly [infer First, ...infer Rest]
+    T extends readonly [infer First extends DeeplyReadonly<Query>, ...infer Rest]
```

Comment:
Won't this break compatibility with older versions of TypeScript?

Reply:
I think you might be right about that, we can probably close this. I'm just going to set `"skipLibCheck: true"` in `tsconfig` instead as a work around